### ISSUE TITLE
NVDA reads aria-describedby on each keystroke

### ISSUE THREAD

**Issue by patrickdemers** [nvaccess/nvda#8796]
### Steps to reproduce:
1. Visit the [characters remaining counter from Deque](https://dequeuniversity.com/library/aria/counters/sf-characters-remaining);
2. Type character in the field "Your message".
### Actual behavior:
A character remaining counter is connected to a textarea with the aria-describedby attribute. NVDA reads the content of the character remaining counter (example : "149 characters remaining of 150 characters") but there is no aria-live yet on the field (when the counter is between 150 and 71 characters).

When the counter is below 71 characters, there is a aria-live attribute added. NVDA reads the counter twice.
### Expected behavior:
NVDA shouldn't read the content of an aria-describedby element the character remaining counter on each keystroke when the counter is between 150 and 71 characters.

Below 71 characters, NVDA should read the counter once.
### System configuration:
Google Chrome version 69.0.3497.100 (Build officiel) (64 bits) (french version)
#### NVDA Installed/portable/running from source:
NVDA portable
#### NVDA version:
NVDA 2018.3.2
#### Windows version:
Windows 10 Entreprise 1703 build 15063.0
#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA?
No

**Comment by jawhien** [comment]
Hello.

I can confirm that I observe the same behavior in google chrome on windows 7.
still more 70 characters nvda can't read the meter, but I reads it anyway. in Firefox and SeaMonkey the number of remaining characters is voiced only when there are less than 71 and once, as I understand it this is correct behavior.

**Comment by Adriani90** [comment]
@patrickdemers I wonder if it is possible for a web developer to implement a sound when maximal number of characters is reached. Tis would be more confortable than NVDA reporting the numbers all the time. NVDA could report the number once when typing the first character and then it could play a tone when the maximum number has been reached.

**Comment by Adriani90** [comment]
@mltony your thoughts are appreciated. Do you think it is possible to implement something like this in general for edit fields with restricted number of characters?

**Comment by mltony** [comment]
Here are my thoughts. It seems there are two separate topics here.
1. The way NVDA reads ARIA attributes. This sounds like a bug in NVDA as NVDA shouldn't be reading this attribute twice. Unfortunately I don't know much how NVDA interact with browsers and about ARIA to provide more insight.
2. How NVDA should be working with edit boxes in general. It seems to me that the vast majority of edit boxes out there have the same predictable behavior. You type letter A, and a split second later this character will actually appear in the text of that edit box in the predictable location, that is right after the previous location of the cursor. But occasionally, that predicted thing is not happening. Limit on the number of characters in the edit box is one such example - the character won't show up. Another similar scenario that comes to my mind is when overwrite mode is accidentally turned on (by pressing Insert key twice) and then you happen to overwrite half of your document before noticing that something has gone wrong. So I think the way we can try to combat both of these issues is to have a predictive model of edit boxes - I appologize for a fancy term, but what I really mean by that is to have a heuristic function, that tries to predict what's going to be the next state of an edit box when a new character is being typed into it. If the predicted state does not match the actual state - that's the time to alert the user with some warning sound signal. We play a warning sound if Caps lock is accidentally turned on already for example. Implementing a predictive edit box model shouldn't be too hard, but maybe it's going to be useful to some.

**Comment by patrickdemers** [comment]
This issue occurs only with Google Chrome, not with Mozilla Firefox.

**Comment by jscholes** [comment]
This is an issue report relating to the use of ARIA on a web page and how NVDA responds to it.  Discussion of how web developers can add sound playback to text fields is out of scope, as it has nothing to do with NVDA whatsoever.

**Comment by Adriani90** [comment]
@patrickdemers it seems the link in the description is outdated. Could you please post an updated test case? Thanks.

**Comment by patrickdemers** [comment]
The characters counter is not available anymore

**Comment by Adriani90** [comment]
Thanks for the swift reply. I am closing this issue for now. Please comment on it if you come accross a new test case and we can reopen it. Thanks.

**Comment by jscholes** [comment]
Here is a working test case for this behaviour: https://jscholes.github.io/aria-describedby_with_focus.html

Related ticket: https://github.com/w3c/aria/issues/1200
